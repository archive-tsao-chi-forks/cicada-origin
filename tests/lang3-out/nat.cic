Nat : Type
Nat = { Zero | Succ }

Zero : Type
Zero = {
  kind : "Zero"
}

Succ : Type
Succ = {
  kind : "Succ"
  prev : Nat
}

zero : Nat
zero = {
  kind = "Zero"
}

succ : (Nat) -> Nat
succ = (prev) => {
  kind = "Succ"
  prev = prev
}

@show zero
@show succ(zero)
@show succ(succ(zero))

// @show {
//   Nat -- {
//     kind = "Succ"
//     prev = {
//       kind = "Succ"
//       prev = zero
//     }
//   }
// }
