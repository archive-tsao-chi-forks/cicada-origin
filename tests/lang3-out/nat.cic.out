{ {
  kind : "Zero"
} | {
  kind : "Succ"
  prev : { {
    kind : "Zero"
  } | {
    kind : "Succ"
    prev : Nat
  } }
} } -- {
  kind = "Zero"
}
{ {
  kind : "Zero"
} | {
  kind : "Succ"
  prev : { {
    kind : "Zero"
  } | {
    kind : "Succ"
    prev : Nat
  } }
} } -- {
  kind = "Succ"
  prev = {
    kind = "Zero"
  }
}
{ {
  kind : "Zero"
} | {
  kind : "Succ"
  prev : { {
    kind : "Zero"
  } | {
    kind : "Succ"
    prev : Nat
  } }
} } -- {
  kind = "Succ"
  prev = {
    kind = "Succ"
    prev = {
      kind = "Zero"
    }
  }
}
